src/Symfony/Component/TypeInfo/Type/CollectionType.php

```diff
@@ -92,6 +92,41 @@ public function wrappedTypeIsSatisfiedBy(callable $specification): bool
         return $this->getWrappedType()->isSatisfiedBy($specification);
     }
 
+    public function accepts(mixed $value): bool
+    {
+        if (!parent::accepts($value)) {
+            return false;
+        }
+
+        if ($this->isList() && !\is_array($value)) {
+            return false;
+        }
+
+        if ($this->isList()) {
+            // only array can be list
+            if (!\is_array($value)) {
+                return false;
+            }
+
+            // array is not a list
+            if (!array_is_list($value)) {
+                return false;
+            }
+        }
```

Comment:
The condition is duplicated. You can factorize like it.

```suggestion
        if ($this->isList() && !(\is_array($value) && array_is_list($value)) {
            return false;
        }
```